src/subgraph/index.ts

```diff
@@ -2,65 +2,89 @@ import { getAllOrdersByUserAddress, getAllPendingOrders, getOrderById, getPythPr
 import { RpcConfig, SubgraphConfig } from '../types';
 import { getAllPositionsByUserAddress, getPositionById } from './positions';
 import { getAllMarketsFromSubgraph, getMarketById } from './markets';
-import { Market, Order, Position } from './common';
+import { Market, Order, Position, getPublicSubgraphEndpoint } from './common';
+import { Chain } from '@parifi/references';
+import { GraphQLClient } from 'graphql-request';
 
 export * from './common';
 export * from './markets';
 export * from './orders';
 export * from './positions';
 
-export class SubGraph {
+export class Subgraph {
+  // The rpcConfig and subgraphConfig objects that are passed to the class will only
+  // be used for the creation of the required fields for initialization
+  // @todo Add authentication to Graph QL Client in a separate PR
+  // Use the below graphQLClient for all requests to the subgraph
+  public graphQLClient: GraphQLClient;
+
   constructor(
     private rpcConfig: RpcConfig,
-    private subGraphConfig: SubgraphConfig,
-  ) {}
+    private subgraphConfig: SubgraphConfig,
+  ) {
+    // Initialize the Graph QL Client using the config received
+    if (subgraphConfig.subgraphEndpoint) {
+      this.graphQLClient = new GraphQLClient(subgraphConfig.subgraphEndpoint);
+    } else {
+      this.graphQLClient = new GraphQLClient(getPublicSubgraphEndpoint(rpcConfig.chainId));
+    }
+  }
 
-  // orders
+  // Returns the configured subgraph endpoint if set, otherwise returns public subgraph endpoint
+  public getSubgraphEndpoint(chainId: Chain): string {
+    if (this.subgraphConfig.subgraphEndpoint) {
+      return this.subgraphConfig.subgraphEndpoint;
+    }
+    return getPublicSubgraphEndpoint(chainId);
+  }
+
+  ////////////////////////////////////////////////////////////////
+  //////////////////////    ORDERS    ////////////////////////////
+  ////////////////////////////////////////////////////////////////
   public async getAllOrdersByUserAddress(userAddress: string, count: number, skip: number): Promise<Order[]> {
-    return await getAllOrdersByUserAddress(
-      this.rpcConfig.chainId,
-      userAddress,
-      this.subGraphConfig.subgraphEndpoint,
-      count,
-      skip,
-    );
+    const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
```

Comment:
same for `this.rpcConfig.chainId` where is this initialized?

Reply:
This is defined in the constructor using a new syntax, which defines and initializes the fields